defaults.properties.yml

```diff
@@ -0,0 +1,30 @@
+build:
+  artifact_mode: symlink
+  test_output: /dev/null
+
+drupal:
+  root: web
+  uri: "https://${projectname}.local"
+  hash_salt: temporary
+  profile: config_installer
+
+  database:
+    database: drupal
+    username: root
+    password: root
+    host: 127.0.0.1
+
+  settings:
+    file_public_path: sites/default/files
+    file_private_path: ""
+
+  twig:
+    debug: false
+
+artifact:
+  directory: artifacts/build
+  git:
+    # The following property is commented out, because this value should only be set in
+    # per-project properties files.
+    # remote:
+      
```

Comment:
This file is used by the installer to set defaults during install, but the primary purpose for this file is to provide global defaults when running normal build tasks -- it's loaded by the `tasks/boilerplate.xml` file when that gets included at the top of every project's `build.xml`.

Since phing uses the first value assigned to a property, rather than the last -- properties are const once declared unless you specifically overwrite them -- the-build loads property files in this order:

* project's `build.ENVIRONMENT.properties.yml`
* project's `build.default.properties.yml`
* the-build's `defaults.properties.yml`

`defaults.properties.yml` is in the project root because that makes it more discoverable for developers -- I think that this will make the available properties and their default values more self documenting, and potentially make it easier for folks to figure out how to customize the build process for specific projects.

Reply:
That makes perfect sense.  Thanks!